inat_finder.py

```
@@ -0,0 +1,589 @@
+    the function extracts and returns that observation number as a string.
+    If the input does not match the URL pattern or is already an observation number,
+    the original string is returned unchanged.
+
+    Args:
+        url_or_number: A string representing either an iNaturalist URL or an observation number.
+
+    Returns:
+        A string containing the extracted observation number, or the original input if no valid
+        observation number is found.
+    """
+    import re
```

Pylint (3.3.7)

[convention] Import outside toplevel (re)

(C0415)